Frameworks/Foundation/NSURLCredential.mm

```diff
@@ -71,6 +75,7 @@ - (instancetype)initWithIdentity:(SecIdentityRef)identity
 - (instancetype)initWithTrust:(SecTrustRef)trust {
     if (self = [super init]) {
         _trust = trust;
+        _persistence = NSURLCredentialPersistenceForSession;
     }
```

Comment:
if(self = [self _initWithPersistence::NSURLCredentialPersistenceForSession]) {
  _trust = trust
}

given we are calling [super init] there anyways.